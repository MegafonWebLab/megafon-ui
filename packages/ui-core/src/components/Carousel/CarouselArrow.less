@import '~styles/base.less';

@b: e('.mfui-carousel-arrow');
@s: e('.slick');

@{b} {
    &__arrow-icon {
        svg {
            display: block;
            width: 40px;
            height: 20px;
            fill: @freshAsphalt;
        }
    }
};

@{s} {
    &-disabled {
        @{b}__arrow-icon {
            svg {
                fill: @black10;
            }
        }
    }
}
