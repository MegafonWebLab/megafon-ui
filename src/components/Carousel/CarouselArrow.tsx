import * as React from 'react';
import * as PropTypes from 'prop-types';
import { cnCreate } from '../../utils/cn';
import './CarouselArrow.less';
import Arrow from 'icons/System/16/Arrow_forward_16.svg';
import ArrowLikeQuote from 'icons/System/16/Arrow_right_16.svg';

interface ICarouselArrowProps {
    className: string;
    onClick: any;
    theme?: string;
}

const cn = cnCreate('mfui-carousel-arrow');
class CarouselArrow extends React.Component<ICarouselArrowProps> {
    static propTypes = {
        className: PropTypes.string,
        onClick: PropTypes.func,
        theme: PropTypes.string,
    };

    renderIcon() {
        const { theme } = this.props;

        return theme === 'showcase' ? <ArrowLikeQuote /> : <Arrow />;
    }

    render() {
        const { className, onClick } = this.props;

        return (
            <div className={cn('', {}, className)} onClick={onClick}>
                <div className={cn('arrow-icon')}>
                    {this.renderIcon()}
                </div>
            </div>
        );
    }
}

export default CarouselArrow;
