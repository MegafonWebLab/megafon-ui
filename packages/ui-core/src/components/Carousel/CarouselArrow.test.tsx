import * as React from 'react';
import { shallow } from 'enzyme';
import CarouselArrow from './CarouselArrow';

describe('<CarouselArrow />', () => {
    describe('render tests', () => {
        it('should render component with all props', () => {
            const wrapper = shallow(
                <CarouselArrow
                    className="some-class"
                    onClick={jest.fn()}
                />
            );

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('onClick tests', () => {
        it('should call onClick on arrow wrapper click', () => {
            const onClick = jest.fn();

            const wrapper = shallow(
                <CarouselArrow
                    onClick={onClick}
                    className="some-class"
                />
            );

            wrapper.find('.mfui-carousel-arrow').simulate('click');

            expect(onClick).toHaveBeenCalledTimes(1);
        });
    });
});
