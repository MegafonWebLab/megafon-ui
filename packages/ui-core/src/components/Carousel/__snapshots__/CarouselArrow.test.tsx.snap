// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarouselArrow /> render tests should render component with all props 1`] = `
<div
  className="mfui-carousel-arrow some-class"
  onClick={[MockFunction]}
>
  <div
    className="mfui-carousel-arrow__arrow-icon"
  >
    <test-file-stub />
  </div>
</div>
`;
